### DISCUSSION TITLE
Refetch only if query is not active

### DISCUSSION THREAD

**Discussion by aliakbarazizi** [TanStack/query#6895] [Q&A]
I wonder is there anyway to refetch only when the query is not active?
I already try this
`refetchOnMount: (query) => !query.isActive(),`

But the value from `isActive` is the fresh value and its always consider active, but I want to know if the query was active before or not.

For explaining why I want this
I have a query which used in many component and in some page all of them are mounting(sometime with small 1 sec delay) and because of that, its trying to refetch the query because its mounted, but since the query is active in another component I don't want to refetch,

I can put `staleTime` to 1 second but I wonder is there any more cleaner to this?

in summary

I just to know to access `isActive` before query mounted.

**Comment by TkDodo** [comment]
staleTime is the solution :)

**Reply by aliakbarazizi** [reply]
The problem with staleTime is it's not good in some cases.
Imagine I have 2 component and both of them use same query.
I want to show fresh data on mount, but If I use one component and show another one in modal I don't want the new data, since the query is currently active. I want to show the same data as I show outside the modal.

I think it's would be nice to have `isActive` value in in refetchOnMount

**Reply by TkDodo** [reply]
> but If I use one component and show another one in modal I don't want the new data

caching is usually time based. If I have one component, leave it open for 5 minutes, and then open the dialog - why wouldn't I want new data?

**Reply by aliakbarazizi** [reply]
Currently, it will not refetch data if data is stale, it will refetch data on mount or other event. so it need a event to trigger,
If open data for more than 5 minute it will not refetch it if I don't trigger any of this.

As I see the reason behind this to have better UX, Now when I open the modal since users already saw the data it's better to show the user same data as the currently active component.

There is huge different for fetch on mount when no component is active and refetch on mount when there is a active component listening on that query, and having option to specify which would be great I think

**Reply by aliakbarazizi** [reply]
For another example, I have tag which user can assign to comment from side bar or from comment body.
I fetch tag for showing them in sidebar
when user click on any comment, I want to show tags in loaded comment as well, and because it will count as refetchOnMount, it will refresh all tags. that's I think is not good for UX. I should not update the data without any event(sometime you can do it by interval but still its a event)
fetching on mount while there is a active query can cause data lost, since data will be refresh and for example user can lose scroll position
